lib/unicode.ml

```diff
@@ -126,26 +126,29 @@ let utf8_of_unicode n =
   if n < 128 then
     String.make 1 (Char.chr n)
   else if n < 2048 then
-    let s = String.make 2 (Char.chr (128 + n mod 64)) in
-    begin
-      s.[0] <- Char.chr (192 + n / 64);
-      s
-    end
+    String.init 2 (fun idx ->
+        match idx with
+        | 0 -> Char.chr (192 + n / 64)
+        | 1 -> Char.chr (128 + n mod 64)
+        | _ -> 'x'
+      )
   else if n < 65536 then
-    let s = String.make 3 (Char.chr (128 + n mod 64)) in
-    begin
-      s.[1] <- Char.chr (128 + (n / 64) mod 64);
-      s.[0] <- Char.chr (224 + n / 4096);
-      s
-    end
+    String.init 3 (fun idx ->
+        match idx with
+        | 0 -> Char.chr (224 + n / 4096)
+        | 1 -> Char.chr (128 + (n / 64) mod 64)
+        | 2 -> Char.chr (128 + n mod 64)
+        | _ -> 'x'
```

Comment:
Actually, if we are willing to use String.init and thus don't care much about performances, we might just as well go the whole way toward higher order. The whole code then becomes:

    if n < 128 then
      String.make 1 (Char.chr n)
    else
      let (m,s) = if n < 2048 then (2,192) else if n < 65536 then (3,224) else (4,240) in
      String.init m (fun i ->
        let j = (n lsr ((m - 1 - i) * 6)) land 63 in
        Char. chr (j + if i = 0 then s else 128))